### ISSUE TITLE
Pass a filter object to user defined predicate in filter2 api

### ISSUE THREAD

**Issue by asfimport** [apache/parquet-java#1607]
Currently for creating a user defined predicate using the new filter api,  no value can be passed to create a dynamic filter at runtime. This reduces the usefulness of the user defined predicate, and  meaningful predicates cannot be created. We can add a generic Object value that is passed through the api, which can internally be used in the keep function of the user defined predicate for creating many different types of filters.
For example, in spark sql, we can pass in a list of filter values for a where IN clause query and filter the row values based on that list.

**Reporter**: [Yash Datta](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=saucam) / @saucam
#### Related issues:
- [Make use of SearchArgument classes for Parquet SERDE](https://issues.apache.org/jira/browse/HIVE-8122) (is blocked by)
- [Pass a filter object to user defined predicate in filter2 api](https://issues.apache.org/jira/browse/PARQUET-116) (duplicates)

<sub>**Note**: *This issue was originally created as [PARQUET-115](https://issues.apache.org/jira/browse/PARQUET-115). Please see the [migration documentation](https://issues.apache.org/jira/browse/PARQUET-2502) for further details.*</sub>

**Comment by asfimport** [comment]
[Julien Le Dem](https://issues.apache.org/jira/browse/PARQUET-115?#comment-14190643) / @julienledem:
Is this a duplicate of PARQUET-116 ?

**Comment by asfimport** [comment]
[Yash Datta](https://issues.apache.org/jira/browse/PARQUET-115?#comment-14191346) / @saucam:
yes, but i am wondering who created this one

**Comment by asfimport** [comment]
[Julien Le Dem](https://issues.apache.org/jira/browse/PARQUET-115?#comment-14194965) / @julienledem:
maybe a refresh in the browser...
